### PR TITLE
Specify Logic Resource.

### PR DESCRIPTION
**Pull request by cqc-alec**

### CODE REVIEW

File: specification/hugr.md [hunk 1]
```diff
@@ -1675,6 +1675,24 @@ implementation of this design document is available on GitHub.
...
+The following operations are defined:
+
+| Name    | Inputs         | Outputs | Meaning                                |
+| ------- | -------------- | ------- | -------------------------------------- |
+| `true`  | none           | `bool`  | const node producing the value "true"  |
+| `false` | none           | `bool`  | const node producing the value "false" |
+| `not`   | `bool`         | `bool`  | logical "not"                          |
+| `and`   | `bool`, `bool` | `bool`  | logical "and"                          |
+| `or`    | `bool`, `bool` | `bool`  | logical "or"                           |
+| `xor`   | `bool`, `bool` | `bool`  | logical "xor"                          |
```
[the remark this answers is not in the document]
Comment (cqc-alec) [reply specification/hugr.md:1694]: Couple of changes I'd like to make here on reflection:
* remove `xor`;
* allow `and` and `or` to take any number of inputs.
(Then `true` is just `and` with 0 inputs and `false` is just `or` with 0 inputs and we just have three basic op types.)

[the remark this answers is not in the document]
Comment (ss2165) [reply specification/hugr.md]: why is `not` not N-ary?

[the remark this answers is not in the document]
Comment (ss2165) [reply specification/hugr.md]: also where do you see operations like `xor` being defined? It's too primitive to be defined as a library function no?

**Review by ss2165 — COMMENTED** [review]
Comments already added as single comments

File: specification/hugr.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (cqc-alec) [reply specification/hugr.md]: Presumably that would mean applying N nots in parallel -- so N inputs and N outputs? Yes we could define this, but it seems less useful. N-ary and and or save us from chaining a bunch of them together in a tree, taking advantage of associativity and commutativity, but I don't think there's a particularly compelling reason to combine parallel nots into one.

[the remark this answers is not in the document]
Comment (cqc-alec) [reply specification/hugr.md]: I don't think any of C, C++ or Python have a _logical_ xor operation (analogous to `&&` or `||`, or `and` and `or`) (though they all have bitwise xor).

[the remark this answers is not in the document]
Comment (ss2165) [reply specification/hugr.md]: that makes sense - happy with current state

**Review by ss2165 — APPROVED** [review]

**Pull request merged**